### ISSUE TITLE
Calling WorldMapFrame:SetMapID() causes taint and prevents certain spells/macros from casting

### ISSUE THREAD

**Issue by fechan** [Stanzilla/WoWUIBugs#352]
## Description
Calls to `WorldMapFrame:SetMapID()` from `/run` or from an addon, then entering and leaving an instanced dungeon will cause subsequent attempts to cast macros or certain spells to fail with the following message:
![image](https://user-images.githubusercontent.com/56131910/202878580-eb8c67f1-6a76-46bb-b300-49f2b94c06d2.png)

This is a regression introduced in 10.0. This causes addons which use the SetMapID function to fail, whereas they worked perfectly in 9.x and below.

## Reproduction steps
1. Create any macro (even an empty one) and put it on Action bar 4
2. Assign it a keybind
3. Run the command, `/run WorldMapFrame:SetMapID(1)`
4. Enter an instanced dungeon (e.g. Ragefire Chasm)
5. Leave the instanced dungeon
6. Press the keybind you assigned to the macro

**Comment by fechan** [comment]
Possibly related to #339

**Comment by p3lim** [comment]
I've previously suggested to a Blizzard employee if we could get an API like the following:
```lua
C_Map.OpenMap([mapID])
C_Map.CloseMap()
```

This would separate map opening/closing invocation completely from the mechanics, both working around this specific issue, but also enabling addons to show/hide the map in combat like we could before. There's a similar pattern with [C_TradeSkillUI.OpenTradeSkill()](https://wowpedia.fandom.com/wiki/API_C_TradeSkillUI.OpenTradeSkill), which is why this is a valid suggestion (although that one requries a hardware event).

When I suggested this I got a positive response, so it might happen.
